### ISSUE TITLE
Setuptools setup.cfg? deprecation

### ISSUE THREAD

**Issue by marscher** [BAMWelDX/weldx#754]
It looks like Setuptools discourages the usage of setup.cfg in favour of pyproject.toml.
I need to validate this, but we get the following deprecation warning during the invocation of setup.py 

```
/setuptools/config/__init__.py:28: SetuptoolsDeprecationWarning: As setuptools moves its configuration towards `pyproject.toml`,
`setuptools.config.read_configuration` became deprecated.

For the time being, you can use the `setuptools.config.setupcfg` module
to access a backward compatible API, but this module is provisional
and might be removed in the future.
```
xref: https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html

**Comment by CagtayFabry** [comment]
good to know, thank you

I guess there is still some time to move over to `pyproject.toml` right?

If possible I would like to wait till Python 3.11 (which afaik should come with a toml parser by default) to move our toolchain over to `pyproject.toml`

**Comment by marscher** [comment]
I think this transition will be carefully treated by the community, as lots of projects use setup.cfg to store their metadata. So we can take our time to make this change.

**Comment by marscher** [comment]
fixed by #774
